File: DotNetNuke/Sample/Presenters/DependencyUC2Presenter.vb
Imports DNN.MVP.Base.BasePresenter
Imports DNN.MVP.Sample.ModuleName1.IViews

Namespace Presenters
    Public Class DependencyUC2Presenter
        Inherits BasePresenter(Of IDependencyUC2View, Object)

        Public Sub New(view As IDependencyUC2View)
            MyBase.New(view)
        End Sub

        Protected Overrides Sub ViewAction()
        End Sub

        Protected Overrides Sub UpdateAction()
        End Sub

        Protected Overrides Sub DeleteAction()
        End Sub

        Protected Overrides Sub InsertAction()
        End Sub

        Protected Overrides Sub InitializeAction()
        End Sub

        Protected Overrides Sub SearchAction()
        End Sub

        Public Overrides Sub Validate()
        End Sub
    End Class
End Namespace

File: MVP.Sample/MVP.Sample.Web/Presenters/DependencyUC2Presenter.cs
using MVP.Base.BasePresenter;
using MVP.Sample.Web.IRepositories;
using MVP.Sample.Web.IViews;
using MVP.Sample.Web.Repositories;

namespace MVP.Sample.Web.Presenters
{
    public class DependencyUC2Presenter : BasePresenter<IDependencyUC2View, IDependencyUC2Repository>
    {
        #region Constructors
        public DependencyUC2Presenter(IDependencyUC2View view)
            : base(view)
        {
            Repository = new DependencyUC2Repository();
        }

        public DependencyUC2Presenter(IDependencyUC2View view, IDependencyUC2Repository repository): base(view, repository)
        {
        }
        #endregion

        #region Overrides of BasePresenter<IDependencyUC2View,object>

        protected override void ViewAction()
        {
        }

        protected override void UpdateAction()
        {
        }

        protected override void DeleteAction()
        {
        }

        protected override void InsertAction()
        {
            //throw new NotImplementedException();
        }

        protected override void InitializeAction()
        {
        }

        protected override void SearchAction()
        {
        }

        public override void Validate()
        {
        }

        #endregion
    }
}